### ISSUE TITLE
multi gpu support

### ISSUE THREAD

**Issue by yhcao6** [dgriff777/rl_a3c_pytorch#24]
When I run the program on multi gpu, that is, I set gpu_id to be [0, 1, 2, 3], it report error "Some of weight/gradient/input teSLTM nsors are located on different GPUs. Please move them to a single one"

**Comment by dgriff777** [comment]
How are you setting it?

A good example of doing so is command line sample on readme 
```
python main.py --env PongDeterministic-v4 --workers 32 --gpu-ids 0 1 2 3 --amsgrad True
```

**Comment by yhcao6** [comment]
<img width="850" alt="screen shot 2018-03-24 at 3 19 39 pm" src="https://user-images.githubusercontent.com/22662425/37861538-d2452614-2f76-11e8-8b30-6144d9f7661d.png">

Still have problem, don't know why, torch version is Version: 0.4.0a0.post2

**Comment by yhcao6** [comment]
If I set all gpu_id to be same for train and test process, then it is okay

**Comment by dgriff777** [comment]
Is your repo current to repo on GitHub? What are you running this on? 
I have never tried running on version 0.4.0a0.post2 yet. May be a bug in pytorch

**Comment by yhcao6** [comment]
Yeah, I just git clone this repo. I think also, maybe it is a bug of pytorch

**Comment by dgriff777** [comment]
yeah try with a more stable version like 0.3 and see if problem still persists
